### DISCUSSION TITLE
Handling long query response lines

### DISCUSSION THREAD

**Discussion by deefrawley** [Flow-Launcher/Flow.Launcher#683] [Ideas]
Currently any returned query line that is longer than the window simply gets cut off which is not an optimal user experience as I can't see any way to see the entire string. Is there a way that core Flow may be able to handle this, by query options so the plugin devs or even users can choose? A few options I can think of:

- Flow breaks long lines into multiple queries, adds something to the title to show it's a continuation (like "cont/..."). Good for information responses but would get messy for things like file/folder locations such as the Explorer plugin

- Implement some sort of scrolling marquee where Flow will animate/scroll the result when it is highlighted in the list. Left, right arrows jump to either end of the result instantly and when focus lost the result shows from the left like today

- Flow breaks the query response when overlength and adds something to show this. So print the first X characters, force something like elipses in (...) then show say the last ten characters. Still lose information like it is now but probably more useful for things where you need to see the final bit like file/folder names.

The scrolling marquee solves all problems but is probably the hardest to implement. Does anyone else have any ideas, thoughts, or things they have seem implemented elsewhere to solve this problem?

**Comment by jjw24** [comment]
may need to see how other similar apps are doing this. Listary is doing marquee scrolling if i remember correctly. 

Currently though, you can hover over a title or subtitle to see their entire string
